File: src/models/tokenizer/ExactSearchTokenizer.java
package models.tokenizer;

import java.util.Arrays;
import java.util.Set;
import java.util.stream.Collectors;

public class ExactSearchTokenizer implements Tokenizer {
    private static final String SPLITTER = "\\s";

    @Override
    public Set<String> tokenize(String text) {
        return Arrays.stream(text.split(SPLITTER)).collect(Collectors.toSet());
    }
}

File: C#/SimpleSearch/SimpleSearch/models/ExactSearchTokenizer.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.RegularExpressions;
using System.Threading.Tasks;

namespace SimpleSearch.models.tokenizer
{
    class ExactSearchTokenizer : Tokenizer
    {
        public override ISet<string> Tokenize(string text) => Regex.Split(text, SPLITTER).ToHashSet();

        public override LinkedList<string> Develope(string token) => new LinkedList<string>((new string[] {token}).ToList());
    }
}